packages/braid-design-system/src/lib/components/Dialog/Dialog.docs.tsx

```diff
@@ -70,13 +103,96 @@ const docs: ComponentDocs = {
     },
   ],
   additional: [
+    {
+      label: 'Title and description',
+      description: (
+        <>
+          <Text>
+            The <Strong>title</Strong> prop provides an accessible name
+            announced to the user when the Dialog is opened. Optionally, a{' '}
+            <Strong>description</Strong> can be provided and will be announced
+            by a screen reader as well as visually forming part of the header
+            block.
+          </Text>
+          <Alert>
+            <Text>
+              Open in Playroom and enable your screen reader to preview the
+              announcements.
+            </Text>
+          </Alert>
+        </>
+      ),
+      background: false,
+      Example: ({ id, getState, toggleState }) => {
+        const { code } = source(
```

Comment:
I'm not very excited about all the duplication. It's very easy for the code and example to drift apart and it's very easy to miss in a review.

Perhaps we can avoid it by using the `flattenChildren` util? Something like this (I tested the code locally):

```tsx
const { code, value } = source(
  <>
    <Inline /* with all its props */ />

    <Dialog /* with all its props */ />
  </>
);
const [toggle, dialog] = flattenChildren(value); // => [<Inline />, <Dialog />];

return {
  code,
  value: (
    <Box borderRadius="xlarge" overflow="hidden">
      <DialogPreview>
        <DialogContent
          {...(dialog as React.ReactElement).props}
          open={() => {}}
          onClose={() => {}}
          scrollLock={false}
        />
        <Screen />
      </DialogPreview>
    </Box>
  ),
};
```

What do you think?

Reply:
Yeah i felt that. I like the work around idea. I'll riff on that a little bit and update 👏

Reply:
Actually, pivoted a bit here. Given the preview is just the Dialog or Drawer, having the Button in the code was confusing. So switching these examples to demo with the `open={true}` prop.

Reply:
Nice 👌